### ISSUE TITLE
Export sucrose from Synechococcus

### ISSUE THREAD

**Issue by djinnome** [PNNL-CompBio/CONCERTO#20]

**Comment by djinnome** [comment]
Make sure that cyano excretes sucrose under co2 + light inputs
Make sure that rhodosporidum is capable of growth on sucrose

**Comment by mcnaughtonadm** [comment]
By setting the `EX_glc__D_e` reaction to 0 in the RT CobraPy medium, the FBA solution returns infeasible. If we then add the `EX_sucr_e` reaction into the RT medium at a bound of 1 (what the `EX_glc__D_e` is normally at). The RT model will grow.

The objective values are:

- With `EX_glu__D_e` : 0.0733
- With `EX_sucr_e` : 0.149
- With both: 0.225

**Comment by PBohutskyi** [comment]
hey. I have a real sucrose export data from batch experiment if want  to validate or constrain the model

**Comment by djinnome** [comment]
For Rhodosporidum toruloides?  If so, that would be awesome, as we currently only have export data for Rhodo in a co-culture

**Comment by PBohutskyi** [comment]
It is content (g/L) of sucrose in axenic S.elongatus when it is growing alone and content in S.elongatus-Rhodo co-culture when they grow together mixed, or when they are separated by membranes

**Comment by djinnome** [comment]
Got it.  At what point in time after inoculation was the measurement taken for each condition (axenic, co-culture mixed, co-culture separated)?  Do we also have measurements of biomass (or a proxy such as OD) taken at the same time as sucrose was measured?

**Comment by djinnome** [comment]
https://github.com/PNNL-CompBio/CONCERTO/issues/20#issuecomment-1451435609

@mcnaughtonadm  getting Rhodo to grow on sucrose solves half the problem. The other half is adding sucrose export to cyano, and ensuring it secretes it.

**Comment by mcnaughtonadm** [comment]
As a point of reference towards this, here is the base optimization through CobraPy of the cyano model.

We have an objective value of: 0.0539 with the following `Uptake` and `Sectretion`

<h4 style="box-sizing: border-box; font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-weight: bold; line-height: 1; color: rgb(0, 0, 0); margin: 2em 0px 0px; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Uptake</h4>

Metabolite | Reaction | Flux | C-Number | C-Flux
-- | -- | -- | -- | --
co2_e | EX_co2_e | 0.7191 | 1 | 30.67%
cobalt2_e | EX_cobalt2_e | 2.758E-07 | 0 | 0.00%
fe3_e | EX_fe3_e | 8.635E-05 | 0 | 0.00%
h_e | EX_h_e | 1.882 | 0 | 0.00%
hco3_e | EX_hco3_e | 1.455 | 1 | 62.07%
leu__L_e | EX_leu__L_e | 0.02836 | 6 | 7.26%
mg2_e | EX_mg2_e | 0.0006043 | 0 | 0.00%
nh4_e | EX_nh4_e | 0.0031 | 0 | 0.00%
no3_e | EX_no3_e | 0.4029 | 0 | 0.00%
photon650_e | EX_photon650_e | 0.0006043 | 0 | 0.00%
photon670_e | EX_photon670_e | 11.7 | 0 | 0.00%
photon690_e | EX_photon690_e | 18.2 | 0 | 0.00%
pi_e | EX_pi_e | 0.023 | 0 | 0.00%
so4_e | EX_so4_e | 0.008394 | 0 | 0.00%
amylose_c | SK_amylose_c | 0.02769 | 0 | 0.00%

<h4 style="box-sizing: border-box; font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-weight: bold; line-height: 1; color: rgb(0, 0, 0); margin: 2em 0px 0px; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Secretion</h4>

Metabolite | Reaction | Flux | C-Number | C-Flux
-- | -- | -- | -- | --
5drib_c | DM_5drib_c | -5.6E-06 | 5 | 26.10%
amob_c | DM_amob_c | -1.411E-06 | 15 | 19.73%
co_c | DM_co_c | -5.701E-05 | 1 | 53.14%
dialurate_c | DM_dialurate_c | -2.758E-07 | 4 | 1.03%
pho_loss_c | DM_pho_loss_c | -0.5849 | 0 | 0.00%
o2_e | EX_o2_e | -3.151 | 0 | 0.00%

The current state of `sucr` within the model is as follows:
```
SUCR: h2o_c + sucr_c --> fru_c + glc__D_c
SPP: h2o_c + suc6p_c --> pi_c + sucr_c
```

**Comment by mcnaughtonadm** [comment]
Current Pull Request waiting for Review and Approval  by either @JamesPino or @djinnome introducing Jupyter Notebook showing how to generate a Synechococcus model capable of exporting sucrose. Moving this task to done, but not closing the issue until PR goes through.

**Comment by mcnaughtonadm** [comment]
Issue solved with #33 , closing issue.
